community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/NodeValueIndexCursorTestBase.java

```diff
@@ -553,11 +613,354 @@ public void shouldFindUpdatedNodeInIndexSeek() throws Exception
+            tx.dataWrite().nodeRemoveLabel( strThree3, label );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE,
+                    IndexQuery.range( prop, "one", true, "three", true ) );
+
+            // then
+            assertFalse( node.next() );
+        }
+    }
+
+    @Test
+    public void shouldNotFindUpdatedNodeInRangeSearch() throws Exception
+    {
+        // Given
+        int label = token.nodeLabel( "Node" );
+        int prop = token.propertyKey( "prop" );
+        CapableIndexReference index = schemaRead.index( label, prop );
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor() )
+        {
+            // when
+            tx.dataWrite().nodeSetProperty( strOne, prop, stringValue( "ett" ) );
+            tx.dataWrite().nodeSetProperty( strThree1, prop, stringValue( "tre" ) );
+            tx.dataWrite().nodeSetProperty( strThree2, prop, stringValue( "tre" ) );
+            tx.dataWrite().nodeSetProperty( strThree3, prop, stringValue( "tre" ) );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE,
+                    IndexQuery.range( prop, "one", true, "three", true ) );
+
+            // then
+            assertFalse( node.next() );
+        }
+    }
+
+    @Test
+    public void shouldFindUpdatedNodeInRangeSearch() throws Exception
+    {
+        // Given
+        int label = token.nodeLabel( "Node" );
+        int prop = token.propertyKey( "prop" );
+        CapableIndexReference index = schemaRead.index( label, prop );
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor() )
+        {
+            // when
+            tx.dataWrite().nodeSetProperty( strOne, prop, stringValue( "ett" ) );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE,
+                    IndexQuery.range( prop, "ett", true, "tre", true ) );
+
+            // then
+            assertTrue( node.next() );
+            assertEquals( strOne, node.nodeReference() );
+        }
+    }
+
+    @Test
+    public void shouldFindSwappedNodeInRangeSearch() throws Exception
+    {
+        // Given
+        int label = token.nodeLabel( "Node" );
+        int prop = token.propertyKey( "prop" );
+        CapableIndexReference index = schemaRead.index( label, prop );
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor() )
+        {
+            // when
+            tx.dataWrite().nodeRemoveLabel( strOne, label );
+            tx.dataWrite().nodeAddLabel( strOneNoLabel, label );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE,
+                    IndexQuery.range( prop, "ett", true, "tre", true ) );
```

Comment:
This test might be working by accident... `strOneNoLabel` has `prop = "one"`, but this test is scanning `[ett,tre]`. I think `strThree1`, `strThree2` and `strThree1` are also in that range, so if we were to change the order of returned results, this test could start failing.